File: src/022/p022.py
import os


def load_names():
    file_path = os.path.join(os.path.dirname(__file__), 'p022_names.txt')
    with open(file_path) as file_handle:
        names_list = file_handle.read().replace('\"', '').split(',')
        return sorted(names_list)


def name_score(name):
    return sum(ord(char) - 64 for char in name)


def solve():
    total_score = 0
    names_list = load_names()
    for index, name in enumerate(names_list):
        score = name_score(name) * (index + 1)
        total_score += score
    return total_score


if __name__ == '__main__':
    result = solve()
    print(result)

File: src/022/p022.lua
function name_score(name)
  local total = 0
  for c in name:gmatch("%u") do
    total = total + c:byte() - string.byte('A') + 1
  end
  return total
end

function load_names()
  local all_names = io.open("p022_names.txt"):read()
  local names = {}
  for name in all_names:gmatch('"(%u+)"') do
    table.insert(names, name)
  end
  table.sort(names)
  return names
end

function solve()
  local total = 0
  local names = load_names()
  for index, name in ipairs(names) do
    local score = name_score(name) * index
    total = total + score
  end
  return total
end

local result = solve()
print(result)
